### PR TITLE
Spark: Improve the test case testEqualityDeleteWithFilter

### PR DESCRIPTION
**Pull request by chenjunjiedada**
The unit test `testEqualityDeleteWithFilter` create the table by itself but not create test data. Looks like the idea is to use the same test data from its base class.

### CODE REVIEW

File: spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java [hunk 1]
```diff
@@ -129,6 +129,18 @@ public StructLikeSet rowSet(String name, Table table, String... columns) {
   public void testEqualityDeleteWithFilter() throws IOException {
     String tableName = "test_with_filter";
     Table table = createTable(tableName, SCHEMA, SPEC);
+
```
Comment (openinx) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java:132]: Why not just use the created `test` table ( in `writeTestDataFile` annotated with a `Before` )  ? then we don't have to write those rows again for this newly created `test_with_filter` table.

Comment (aokolnychyi) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java:132]: +1 to what @openinx suggests. Then we also don't need to reason about dropping the table.

**Comment by openinx** [comment]
LGTM, +1

**Comment by chenjunjiedada** [comment]
Some Flink unit tests failed.  Could you re-run the CI? I can close and reopen the PR if necessary.

**Comment by openinx** [comment]
I don't think this patch will affect the flink unit tests. Got this merged now, will take a look to the flink unit tests.

**Pull request merged**